lib/ansible/modules/storage/netapp/na_ontap_interface.py

```diff
@@ -313,15 +319,19 @@ def validate_create_parameters(self, keys):
 
     def create_interface(self):
         ''' calling zapi to create interface '''
-        required_keys = set(['role', 'address', 'home_port', 'netmask'])
-        data_protocols_obj = self.set_protocol_option(required_keys)
+        required_keys = set(['role', 'home_port'])
+        data_protocols_obj = None
+        if self.parameters.get('subnet_name') is None:
+            required_keys.add('address')
+            required_keys.add('netmask')
+            data_protocols_obj = self.set_protocol_option(required_keys)
```

Comment:
Indenting this line appears to have broken setting the protocol for the interface.

I used this exact file on my ansible NetApp playbook to create an interface, and the interface was created with the default protocols each time I tried, despite me specifying "iscsi" specifically.  I also tested with "nfs" specified, and that was ignored as well.

On my own copy, I decreased the indent of this line (pulling it outside the "if" block), and that fixed the issue for my particular use case.  I haven't tested it for other use cases though.

I can write a separate bug report if that's the appropriate way to report this.